apis/container/v1beta1/gkecluster_referencers_test.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	runtimev1alpha1 "github.com/crossplaneio/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplaneio/crossplane-runtime/pkg/resource"
+	"github.com/crossplaneio/crossplane-runtime/pkg/test"
+)
+
+const (
+	mockName          = "mockName"
+	mockGKEClusterURI = "projects/mockProj/locations/mockLoc/clusters/mockName"
+)
+
+var (
+	errBoom = errors.New("boom")
+)
+
+type mockReader struct {
+	client.Reader
+	readFn func(ctx context.Context, key client.ObjectKey, obj runtime.Object) error
+}
+
+func (m *mockReader) Get(ctx context.Context, key client.ObjectKey, obj runtime.Object) error {
+	return m.readFn(ctx, key, obj)
+}
+
+func TestGKEClusterURIReferencerGetStatus(t *testing.T) {
+
+	errResourceNotFound := &kerrors.StatusError{ErrStatus: metav1.Status{Reason: metav1.StatusReasonNotFound}}
+
+	readyResource := GKECluster{
+		Status: GKEClusterStatus{
+			AtProvider: GKEClusterObservation{
+				SelfLink: URIPrefix + mockGKEClusterURI,
+			},
+		},
+	}
+
+	readyResource.Status.SetConditions(runtimev1alpha1.Available())
+
+	type input struct {
+		readerFn func(ctx context.Context, key client.ObjectKey, obj runtime.Object) error
+	}
+	type expected struct {
+		statuses []resource.ReferenceStatus
+		err      error
+	}
+	for name, tc := range map[string]struct {
+		input    input
+		expected expected
+	}{
+		"ReaderError_ReturnsError": {
```

Comment:
I personally can't stand these test names that are _both_ CamelCase and snake_case. :( I feel that typically they're used when we're trying to use the test case names to convey the expectations of the test. I've been experimenting with including a `reason` string in the case that is printed when the test fails, thus allowing a more succinct test name.